src/Sustainsys.Saml2/Xml/XmlTraverser.cs

```diff
@@ -282,10 +282,17 @@ public bool EnsureName(string localName, string namespaceUri)
     /// Ensures that the contents of the current node is only text and returns the text.
     /// </summary>
     /// <returns></returns>
-    public string GetTextContents()
+    public string? GetTextContents()
     {
         //TODO: Test case + error handling for non text content.
         IgnoreChildren();
+        
+        var nullNodeValue = CurrentNode?.Attributes?.GetNamedItem("xsi:nil")?.Value;
```

Comment:
The support for null is specific to attribute values. The [Saml Core Spec](https://groups.oasis-open.org/higherlogic/ws/public/download/56776/sstc-saml-core-errata-2.0-wd-07.pdf/latest) section 2.7.3.1.1 states that:

> If a SAML attribute includes an empty value, such as the empty string, the corresponding
<AttributeValue> element MUST be empty (generally this is serialized as <AttributeValue/>).
This overrides the requirement in Section 1.3.1 that string values in SAML content contain at least one
non-whitespace character.
If a SAML attribute includes a "null" value, the corresponding <AttributeValue> element MUST be
empty and MUST contain the reserved xsi:nil XML attribute with a value of "true" or "1".

Checking 1.3.1 it states that string values must have at least one non-whitespace character.

I think that the correct validation to do in this method is to check that there is indeed at least one character, possibly with an "bool allowEmpty" param. Then the check for xsi:nil can be moved to `ReadElements(XmlTraverser source, SamlAttribute attribute)`.

And btw, the retrieval of the attribute should use the namespace. The prefix `xsi` is just a common thing to do, it doesn't have to be exactly that prefix.

`source.CurrentNode!.Attributes!["nil", XmlSchema.InstanceNamespace]?.Value`